# A Market Analysis for App Store and Google Play



This project aims to figure out profitable marketing personas and profiles for the App Store and Google Play digital stores. Marketing teams will use this information for their marketing efforts. We want to help our developers understand what type of apps are likely to attract more users on Google Play and the App Store. We're aiming for the English-speaking market for this analysis. Our primary source of revenue is in-app purchases and in-app ads. We're only looking for free to download apps. Therefore the number of users using the app is very important to us.


## The Datasets

In September 2018, both the App Store and Play Store had approximately 2 million apps available. The datasets we're using are from that time. Since four million rows are too big for our resources, we're using a sample of them dwindling them down to 10k rows.

You can download the both [Google Play (data collected in August 2018)](https://www.kaggle.com/lava18/google-play-store-apps) and [App Store (data collected in July 2017)](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) datasets for your use and examination using the links provided.


We'll start by opening the datasets.

In [1]:
"""
This is a very basic approach to analysis in Python.
This project won't use the popular data science libraries like numpy, pandas or matplotlib.
We're aiming to get a grasp for the most basic techniques used in both programming and analysis.
"""

from csv import reader

# Open Appstore.csv
opened_file = open('AppleStore.csv', encoding='utf8')
read = reader(opened_file)
appstore = list(read)
appstore_header = appstore[0]
appstore = appstore[1:]
           
# Open Google Play Data
opened_file = open('googleplaystore.csv', encoding='utf8')
read = reader(opened_file)
playstore = list(read)
playstore_header = playstore[0]
playstore = playstore[1:]

In [2]:
# Exploring the dataset with a pre-written function
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    # Take a slice of the dataset
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns: # if parameter is True, print the lengths.
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        

print('Exploring the AppStore Data')
print('\n')

print(appstore_header)
print('\n')

explore_data(appstore, 0, 5, True)


Exploring the AppStore Data


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number 

## Data Cleaning

In [3]:
print('Exploring the PlayStore Data')    
print('\n')
print(playstore_header)
print('\n')
explore_data(playstore, 0, 5, True) # Using the previous function to explore

Exploring the PlayStore Data


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M'

Our data seems ok. But we have too many rows in our datasets. How can we be sure there aren't problems with the rest of them? We have around 17.000 rows, and for all intents of purpose, there can be all kinds of issues in them. With our limited tools of basic python programming, we're going to try and find the rows with missing values by iterating over both of them.

In [4]:
# Rows with missing information

for row in appstore:
    
    if len(row) != len(appstore[0]): # if the length of any of the rows in the dataset isn't the same as our problem free row
        print(appstore[0])
        print(len(appstore[0]))
        print(row)
        print(len(row)) # print it's information

In [5]:
# Rows with missing information

for row in playstore:
    
    if len(row) != len(playstore[0]):
        
        print(playstore[0]) # Dataset header
        print(len(playstore[0])) # The length of the header (problem free row).
        
        print(row) # The row containing the mistake.
        print(len(row)) # The length of the shorter row.
        
        print(playstore.index(row)) # The number of the row containing the mistake.

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
13
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
12
10472


In [6]:
playstore[10472]

['Life Made WI-Fi Touchscreen Photo Frame',
 '1.9',
 '19',
 '3.0M',
 '1,000+',
 'Free',
 '0',
 'Everyone',
 '',
 'February 11, 2018',
 '1.0.19',
 '4.0 and up']

This row has a blank column in it. We can drop it with the del command.

In [7]:
del playstore[10472]

It's gone now.

In [8]:
playstore[10472]

['osmino Wi-Fi: free WiFi',
 'TOOLS',
 '4.2',
 '134203',
 '4.1M',
 '10,000,000+',
 'Free',
 '0',
 'Everyone',
 'Tools',
 'August 7, 2018',
 '6.06.14',
 '4.4 and up']

The discussion at Kaggle confirms that the App Store database has no errors in itself. So there is no further cleaning needed before starting to analyze the dataset.

## Duplicate entries

But before that, we're going to have to check if there are duplicate entries in these datasets.

In [9]:
duplicate_apps = []
unique_apps = []

for app in playstore:                    # For every row
    
    name = app[0]                        # We extract the name of the app
    
    if name in unique_apps:              # If it's in the list
        duplicate_apps.append(name)      # We append to the duplicated list
        
    else:
        unique_apps.append(name)         # If it's not, we append to unique apps list.
        
print('number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])
print('\n')

number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


The dataset has duplicate entries and has plenty of them. It doesn't need to count apps more than once when we analyze data, so we need to remove the duplicates and keep single entries per app. We could remove duplicate rows, making the process a bit random, but we're going to look for a better way. We're going to start by examining the rows for the Instagram app.

In [10]:
for app in playstore:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


The dataset contains four entries for the Instagram app, and the only column that varies is the review counts column. Which means the one with the higher count is the most recent data of all four. So we're going to keep it and remove others.

In [11]:
reviews_max = {}

for app in playstore: # For every row
    
    name = app[0] # We extract the app's name
    n_reviews= float(app[3]) # We extract the number of reviews
    
    if (name in reviews_max) and (reviews_max[name] < n_reviews): # If the name's in dictionary and has more reviews
        reviews_max[name] = n_reviews # We take it as our data
        
    if name not in reviews_max: # if not in the list
        reviews_max[name] = n_reviews # we add it
        
len(reviews_max)

9659

The code above works, and it works precisely because we don't include an else statement. The code updates the entry ONLY if the review count is lower than the entry at hand. And it adds a key to the dictionary if the name of the entry is not in it. Two if statements that deal with the two conditions we're interested in.

This dictionary contains only the most recent entries of the apps we have in our `playstore` database. We can use it to remove duplicate entries.

In [12]:
android_clean = [] # Our new list
already_added = [] # The list we'll throw the duplicates at

for app in playstore: 
    
    name = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == reviews_max[name]) and (name not in already_added): # If it's the one in our reviews_max, we'll take it
        android_clean.append(app)
        already_added.append(name)
        
print(len(android_clean))
android_clean[:5]

9659


[['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up'],
 ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
  'ART_AND_DESIGN',
  '4.7',
  '87510',
  '8.7M',
  '5,000,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'August 1, 2018',
  '1.2.4',
  '4.0.3 and up'],
 ['Sketch - Draw & Paint',
  'ART_AND_DESIGN',
  '4.5',
  '215644',
  '25M',
  '50,000,000+',
  'Free',
  '0',
  'Teen',
  'Art & Design',
  'June 8, 2018',
  'Varies with device',
  '4.2 and up'],
 ['Pixel Draw - Number Art Coloring Book',
  'ART_AND_DESIGN',
  '4.3',
  '967',
  '2.8M',
  '100,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design;Creativity',
  'June 20, 2018',
  '1.1',
  '4.4 and up'],
 ['Paper flowers instructions',
  'ART_AND_DESIGN',
  '4.4',
  '167',
  '5.6M',
  '50,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'March 26, 2017

The length is what we expected: 9659

Our process was a success. We've removed the duplicate entries. We are now going to make sure that all the apps in the dataset are for our target audience. We work for a company that develops in English, and our efforts directed towards an English-speaking audience. Now let's check the app names for what we call abnormalities in the form of nonlatin alphabet letters.

In [13]:
print(appstore[813][1])
print(appstore[6731][1])
print('\n')
print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング
لعبة تقدر تربح DZ


Here are they! These characters are not Latin-alphabet names, so they aren't targeted towards our English-speaking audience. To remove them from our database, we're going to have to work with ASCII codes.

For example, if you Alt + 68 on any text editor you're going to have: D (upper and lower are different). These are codes on how the machine side of our computers uses. ASCII stands for American Standard Code for Information and is limited to 127, which means any letter with codes above 127 aren't any use to us.

Let's write a function that returns False or True depending on the character's code so we can find use it to find unwanted apps later.

In [14]:
def isenglish(string): # Take a string
    
    for characters in string: # Iterate over every single character in it
        
        if ord(characters) > 127: # If it's code is above 127
            return False  # Return false
        
    return True

print(isenglish('Instagram'))
print(isenglish('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(isenglish('Docs To Go™ Free Office Suite'))
print(isenglish('Instachat 😜'))

True
False
False
False


The result is problematic. The third and fourth apps are English but still returns a False. To not lose useful data, we are going to need a workaround. A quick fix that pops in my mind is to check if the app name has more than three non-ASCII characters. The result will secure emojis, and other extra characters won't throw off our function.

In [15]:
def isenglish(string):
    noneng = 0 # A variable to count nonlating characters
    
    for characters in string:
        
         if ord(characters) > 127: # If it's above 127
            noneng += 1 # Add to the count
            
    if noneng > 3: # If the number is more than 3
        return False
    else:
        return True
    
print(isenglish('Instagram'))
print(isenglish('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(isenglish('Docs To Go™ Free Office Suite'))
print(isenglish('Instachat 😜'))

True
False
True
True


Voila! This works! Now lets use it in our datasets.

In [16]:
playstore_english = [] # Our new list for Play Store

for apps in android_clean:
    name = apps[0]
    if isenglish(name):
        playstore_english.append(apps)
        
appstore_english = [] # Our new list for App Store
for apps in appstore:
    name = apps[1]
    if isenglish(name):
        appstore_english.append(apps)

print('English named Play Store apps ' + str(len(playstore_english)))
print('English named IOS App Store apps ' + str(len(appstore_english)))

English named Play Store apps 9614
English named IOS App Store apps 6183


Our firm develops games that are free to download and focus on in-app revenues through ads. Another subsetting we need to implement is to filter the apps so that we're only left with the free to download apps.

In [17]:
appstore_free = []
playstore_free = []

for apps in playstore_english:
    price = apps[7] # Extract the price from the row
    if price == '0': # if it's 0
        playstore_free.append(apps) # take it

for apps in appstore_english:
    price = apps[4]
    if price == '0.0':
        appstore_free.append(apps)
    

print('Free to download Play Store apps ' + str(len(playstore_free)))
print('Free to download IOS App Store apps ' + str(len(appstore_free)))    

Free to download Play Store apps 8864
Free to download IOS App Store apps 3222


## Data Analysis

Having finished cleaning the dataset we can now focus on analyzing the data.

As mentioned in the introduction, we aim to determine the kinds of apps that are likely to attract more users because our revenue is from the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful in both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

In [18]:
playstore_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [19]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for rows in dataset:
        total += 1
        content = rows[index]
        if content in table:
            table[content] += 1
        else:
            table[content] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
    return table_percentages
        
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
    
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [20]:
display_table(appstore_free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Now, we have our frequency table on free English Appstore apps. We see that most of them (more than half) are Games at %58. It's followed by Entertainment (%7.8) and Photo & Video (%4.96). With Education (%3.7) and Social Networking (%3.3) being the other two in our Top 5 we can conclude that Appstore (at least in free-English filters) populated by `fun` applications rather than practicals such as education, productivity, shopping, lifestyle.

This step doesn't conclude our discussion since just because there are more apps in one category doesn't mean there are more users who are using them. The demand might not be the same as the offer.

Let's continue by having a look at the Google Play Store.

In [21]:
display_table(playstore_free, 1) # Category column

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The trend changes drastically in Google Play Store. Games are the only fun app in the most popular categories at %9.8. While Family, Tools, Productivity, Finance, Business, Life Style category dominates the list. The problem with this is though arbitrary names such as Family and Business.

What does the category Family contain? It turns out its mostly games for kids!



![title](https://camo.githubusercontent.com/9bf24b9efc3d88a3d55f5c09e314987941f0bab5/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f64712d636f6e74656e742f3335302f7079316d385f66616d696c792e706e67)


Still, the practical apps have more presence in Google Playstore compared to the Appstore.

In [22]:
display_table(playstore_free, -4) # Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

`Genres` column is more in detail. The data are more divided and have more defining variables. Since we have that from our `Category` column, this became obsolete.

What do we know?

We now know that fun apps dominate Appstore and practical ones. Play Store with Games is still relevant and possibly the most popular category in Playstore as well - although nowhere near overlooking as it is in Appstore.

Does this correlate to where the users are? Which apps are the most popular, have the most user count?

How do you measure where the users are? The installation numbers are a good measure. For the Google Play data set, we can find this information in the `Installs` column, but this information is missing for the App Store data set.

To work around this, we're going to use the count of the user's reviews of the apps. `rating_count_tot` is the column we're looking for. We will start by calculating the average number of user ratings per app genre on the App Store.

In [23]:
genres_ios = freq_table(appstore_free, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for rows in appstore_free:
        genre_app = rows[-5]
        if genre_app == genre:
            n_ratings = float(rows[5])
            total += n_ratings
            len_genre += 1
    avg_user_rating = total / len_genre
    print(genre, ':', avg_user_rating)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Let's take a closer look at genres with highest number of ratings:

In [24]:
for app in appstore_free:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


As you can see, a couple of apps are dominating the charts. So it isn't a trend but the domination of a couple of apps. The same applies to Social Networking group where apps like Facebook, Pinterest, Skype account for almost of the share. We can deal with this by setting a threshold for the number of ratings and exclude them from our analysis. But we won't go into such a level of detail now.

On the other hand, there is the `Reference` genre:

In [25]:
for app in appstore_free:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5]) # print name and number of ratings

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


## Our take on App Store

We have ourselves a catch right here. References category has somewhat of an excellent distribution. You can argue this is as well a monopoly of some popular books in the form of religious texts and dictionaries. But religious text apps can be copied, and there can be lots of dictionaries for slangs.

Besides those two big scopes of ideas, the market is full of life. There are references for games, pieces of music, there may be for technical fields and Wikia like pages for mangas and animes.

Audiobooks and references for popular hits like Game of Thrones come to mind, Lord of the Rings and Harry Potter are only the other two pops up to mind.

This general picture confirms as well our conception that the fun side of things dominates the App Store. The result is expected since we're continuing our analysis with the same dataset, however.

The other genres that might be used to us are weather, book, drinks, and food.

Weather apps are a possibility but must be very detailed and perhaps show historical data to keep users' interest high to achieve long app-use times. Weather apps, in most cases, are used in short bursts. People take a quick look at what the weather is going to be and turn the app off. So these shortcomings need some work.

Food and drink apps are broad in scope. Having an actual delivery system won't work as it's a very competitive scene, and you can't just as quick startup from zero. Recipe apps can be fun, and people will spend time on them, especially with personalization.

Now, onto Google Play Store.

Google Play dataset has installed columns, so we should be able to get a clearer picture of genre popularity. However, the install numbers don't seem precise enough -- we can see that most values are categorical.

In [26]:
display_table(playstore_free, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We don't know whether an app with 100,000+ installs has 100 or 200 or 300k installs. For our purposes, we don't need our data to be this precise, however. Aggravations such as this scale are just as good as it is enough to give a clear picture.

However, to compute the numbers, we're going to have to convert the values from str to float. In our steps, we're going to have to remove the commas and, plus characters, then cast them as a float.

In [27]:
cats_play = freq_table(playstore_free, 1)

for category in cats_play:
    total  = 0
    len_category = 0
    for row in playstore_free:
        category_app = row[1]
        if category_app == category:
            n_installs = row[5]
            n_installs = n_installs.replace('+', '')
            n_installs = float(n_installs.replace(',', ''))
            total += n_installs
            len_category += 1
    avg_n = total / len_category
    print(category, ':' , avg_n)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Communication apps seems to have the edge over here. Let's take a closer look:

In [28]:
for app in playstore_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Some apps like WhatsApp, Google Chrome, Skype, and Messenger seem to dominate here. So for taking a better look at how this genre holds itself in the market, let's pick an arbitrary threshold for two things. First, remove the big guns from the list. Second, preserve the successful apps in the market.

I think the 100.000.000 is a good pick here. Let's take what do we have if we remove every app that has 100.000.000 and above from our list.

In [29]:
under_100m = []

for app in playstore_free:
    n_installs = app[5]
    n_installs = n_installs.replace(',','')
    n_installs = n_installs.replace('+','')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
        
sum(under_100m) / len(under_100m)
        

3603485.3884615386

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out that this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation.

Let's take a look at the genre we've discovered previously for App Store. The books and references genre. The category name for it is: 'BOOKS_AND_REFERENCE:

In [30]:
for app in playstore_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0],' :', app[5])

E-Book Read - Read Book for free  : 50,000+
Download free book with green book  : 100,000+
Wikipedia  : 10,000,000+
Cool Reader  : 10,000,000+
Free Panda Radio Music  : 100,000+
Book store  : 1,000,000+
FBReader: Favorite Book Reader  : 10,000,000+
English Grammar Complete Handbook  : 500,000+
Free Books - Spirit Fanfiction and Stories  : 1,000,000+
Google Play Books  : 1,000,000,000+
AlReader -any text book reader  : 5,000,000+
Offline English Dictionary  : 100,000+
Offline: English to Tagalog Dictionary  : 500,000+
FamilySearch Tree  : 1,000,000+
Cloud of Books  : 1,000,000+
Recipes of Prophetic Medicine for free  : 500,000+
ReadEra – free ebook reader  : 1,000,000+
Anonymous caller detection  : 10,000+
Ebook Reader  : 5,000,000+
Litnet - E-books  : 100,000+
Read books online  : 5,000,000+
English to Urdu Dictionary  : 500,000+
eBoox: book reader fb2 epub zip  : 1,000,000+
English Persian Dictionary  : 500,000+
Flybook  : 500,000+
All Maths Formulas  : 1,000,000+
Ancestry  : 5,000,00

This genre, as well as it's parallel on App Store, includes a large variety of apps: software for processing and reading ebooks, libraries, dictionaries, audiobooks, tutorials, language-based apps.

Let's look at how the dominating forces of this category are:

In [31]:
for app in playstore_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '100,000,000+' or
                                            app[5] == '500,000,000+' or
                                            app[5] == '1,000,000,000+'):
        print(app[0],'',app[5])

Google Play Books  1,000,000,000+
Bible  100,000,000+
Amazon Kindle  100,000,000+
Wattpad 📖 Free Books  100,000,000+
Audiobooks from Audible  100,000,000+


The dominating forces are rather few in numbers. The result is looking good. Let's take a look at successful apps that stands in the middle.

In [32]:
for app in playstore_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+' or
                                            app[5] == '5,000,000+' or
                                            app[5] == '50,000,000+' or
                                            app[5] == '10,000,000+'):
        print(app[0],'',app[5])

Wikipedia  10,000,000+
Cool Reader  10,000,000+
Book store  1,000,000+
FBReader: Favorite Book Reader  10,000,000+
Free Books - Spirit Fanfiction and Stories  1,000,000+
AlReader -any text book reader  5,000,000+
FamilySearch Tree  1,000,000+
Cloud of Books  1,000,000+
ReadEra – free ebook reader  1,000,000+
Ebook Reader  5,000,000+
Read books online  5,000,000+
eBoox: book reader fb2 epub zip  1,000,000+
All Maths Formulas  1,000,000+
Ancestry  5,000,000+
HTC Help  10,000,000+
Moon+ Reader  10,000,000+
English-Myanmar Dictionary  1,000,000+
Golden Dictionary (EN-AR)  1,000,000+
All Language Translator Free  1,000,000+
Aldiko Book Reader  10,000,000+
Dictionary - WordWeb  5,000,000+
50000 Free eBooks & Free AudioBooks  5,000,000+
Al-Quran (Free)  10,000,000+
Al Quran Indonesia  10,000,000+
Al'Quran Bahasa Indonesia  10,000,000+
Al Quran Al karim  1,000,000+
Al Quran : EAlim - Translations & MP3 Offline  5,000,000+
Koran Read &MP3 30 Juz Offline  1,000,000+
Hafizi Quran 15 lines per pag

We have ourselves a catch.

Quran examples show that popular books work, as well as the Bible. This result parallels our findings for App Store as well. Previous ideas about basing new apps on recently popular books like Lord of the Rings, Harry Potter, Game of Thrones, Twilight series is a good idea. Not only that, this, for example, are all fantasy books. There's a pattern here in itself.

Also, since the same kind of app saturates the stores, we should add more unique features in the form of perhaps an audiobook, daily quotes, forums/message boards, and quizzes about the book. We aim to increase the in-app time as much as people where our revenues are coming.

But that is a whole other matter of subject itself.

### Conclusion

The data about Google Playstore and Apple's App Store was analyzed in this project to recommend an app profile that can be profitable for both markets.

We concluded that references and book genres seem to be the best option that developers can take right now. It has high traffic and low competition when compared to other genres.